dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -21,33 +15,75 @@
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
+  -- Having Iceberg in v2 engine not able to do CTAS makes it complex. Hard to factorize with table materialization
+  -- because of statement block.
+  {%- if format == 'iceberg' -%}
+    {%- set iceberg_tmp_relation = make_temp_relation(target_relation, '__incremental_iceberg_ctas') -%}
+  {%- endif -%}
+
   {% set to_drop = [] %}
   {% if existing_relation is none %}
-      {% set build_sql = create_table_as(False, target_relation, sql) %}
+    {%- if format == 'iceberg' -%}
+      {%- set build_sql = create_table_iceberg(target_relation, existing_relation, iceberg_tmp_relation, sql) -%}
+      {% do to_drop.append(iceberg_tmp_relation) %}
+    {% else %}
+      {% set build_sql = create_table_as(False, target_relation, sql) -%}
+    {%- endif -%}
   {% elif existing_relation.is_view or should_full_refresh() %}
-      {% do adapter.drop_relation(existing_relation) %}
-      {% set build_sql = create_table_as(False, target_relation, sql) %}
+    {% do adapter.drop_relation(existing_relation) %}
+    {%- if format == 'iceberg' -%}
+      {%- set build_sql = create_table_iceberg(target_relation, existing_relation, iceberg_tmp_relation, sql) -%}
+      {% do to_drop.append(iceberg_tmp_relation) %}
+    {% else %}
+      {% set build_sql = create_table_as(False, target_relation, sql) -%}
+    {%- endif -%}
   {% elif partitioned_by is not none and strategy == 'insert_overwrite' %}
       {% set tmp_relation = make_temp_relation(target_relation) %}
       {% if tmp_relation is not none %}
-          {% do adapter.drop_relation(tmp_relation) %}
+        {% do adapter.drop_relation(tmp_relation) %}
       {% endif %}
       {% do run_query(create_table_as(True, tmp_relation, sql)) %}
       {% do delete_overlapping_partitions(target_relation, tmp_relation, partitioned_by) %}
       {% set build_sql = incremental_insert(tmp_relation, target_relation) %}
       {% do to_drop.append(tmp_relation) %}
-  {% else %}
+  {% elif strategy == 'append' %}
       {% set tmp_relation = make_temp_relation(target_relation) %}
       {% if tmp_relation is not none %}
-          {% do adapter.drop_relation(tmp_relation) %}
+        {% do adapter.drop_relation(tmp_relation) %}
       {% endif %}
-      {% do run_query(create_table_as(True, tmp_relation, sql)) %}
+      {%- if format == 'iceberg' -%}
+         -- We need to use the create_tmp_table_iceberg macro (parquet table) because create_table_as create a table
+         -- with the format inherited from config. For Iceberg, what we are doing is that we create a parquet table
+         -- containing the result of the incremental query (no limit), and then we insert this in the iceberg table.
```

Comment:
I'm wondering if you need to create an iceberg table as tmp table containing the increment or if we can simply create a parquet table and then merge from there? 
The only reason why we should create a tmp iceberg table could be keeping the data types.